src/server/utils/next-code.ts

```
@@ -0,0 +1,71 @@
+export async function getNextCode(
+  model: { findFirst: (args: unknown) => Promise<{ code: number } | null> },
+  companyId: string,
+  options?: {
+    maxRetries?: number;
+    codeField?: string;
+  },
+): Promise<number> {
+  const maxRetries = options?.maxRetries ?? 3;
+  const codeField = options?.codeField ?? "code";
+
+  for (let attempt = 0; attempt <= maxRetries; attempt++) {
```

Biome (2.3.14)

[error] This code will never be reached ...

... because this statement will return from the function beforehand


(lint/correctness/noUnreachable)